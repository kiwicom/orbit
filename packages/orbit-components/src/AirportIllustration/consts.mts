export const NAMES = [
  "BGYFastTrack",
  "BUDFastTrack",
  "MRSSmartPass",
  "NCEFastTrack",
  "PRGSmartPass",
  "VCESmartPass",
];
