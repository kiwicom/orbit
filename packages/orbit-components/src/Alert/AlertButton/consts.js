// @flow
export const TYPE_OPTIONS = {
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
  INFO_SUBTLE: "infoSubtle",
  SUCCESS_SUBTLE: "successSubtle",
  WARNING_SUBTLE: "warningSubtle",
  CRITICAL_SUBTLE: "criticalSubtle",
  SECONDARY: "secondary",
};

export const TOKENS = {
  backgroundButton: "backgroundButton",
  backgroundButtonHover: "backgroundButtonHover",
  backgroundButtonActive: "backgroundButtonActive",
  backgroundButtonFocus: "backgroundButtonFocus",
  colorTextButton: "colorTextButton",
  borderColorButtonFocus: "borderColorButtonFocus",
};
