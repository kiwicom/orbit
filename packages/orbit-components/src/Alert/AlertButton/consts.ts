export enum TYPE_OPTIONS {
  INFO = "info",
  SUCCESS = "success",
  WARNING = "warning",
  CRITICAL = "critical",
  INFO_SUBTLE = "infoSubtle",
  SUCCESS_SUBTLE = "successSubtle",
  WARNING_SUBTLE = "warningSubtle",
  CRITICAL_SUBTLE = "criticalSubtle",
  SECONDARY = "secondary",
}

export enum TOKENS {
  backgroundButton = "backgroundButton",
  backgroundButtonHover = "backgroundButtonHover",
  backgroundButtonActive = "backgroundButtonActive",
  backgroundButtonFocus = "backgroundButtonFocus",
  colorTextButton = "colorTextButton",
  borderColorButtonFocus = "borderColorButtonFocus",
}
