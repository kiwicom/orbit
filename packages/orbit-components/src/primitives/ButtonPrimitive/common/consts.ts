export enum SIZE_OPTIONS {
  SMALL = "small",
  NORMAL = "normal",
  LARGE = "large",
}

export enum TOKENS {
  heightButton = "heightButton",
  fontSizeButton = "fontSizeButton",
  paddingButton = "paddingButton",
  paddingButtonWithIcons = "paddingButtonWithIcons",
  paddingButtonWithLeftIcon = "paddingButtonWithLeftIcon",
  paddingButtonWithRightIcon = "paddingButtonWithRightIcon",
  spinnerWidth = "spinnerWidth",
  spinnerHeight = "spinnerHeight",
  marginRightIcon = "marginRightIcon",
}

export enum BUTTON_STATES {
  DEFAULT = "default",
  HOVER = "hover",
  ACTIVE = "active",
  FOCUS = "focus",
}
