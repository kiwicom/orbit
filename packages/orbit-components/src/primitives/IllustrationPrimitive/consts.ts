import { Size } from "./index.d";

type SizeKeys = "EXTRASMALL" | "SMALL" | "MEDIUM" | "LARGE" | "DISPLAY";

export const SIZE_OPTIONS: Record<SizeKeys, Size> = {
  EXTRASMALL: "extraSmall",
  SMALL: "small",
  MEDIUM: "medium",
  LARGE: "large",
  DISPLAY: "display",
};

export const baseURL = "//images.kiwi.com";
