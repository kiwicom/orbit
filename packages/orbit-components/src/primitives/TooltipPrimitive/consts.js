// @flow

export const SIZE_OPTIONS = {
  SMALL: "small",
  MEDIUM: "medium",
};

// the default order on RTL needs to be changed
export const RTL_POSITIONS = {
  LEFT: "left",
  RIGHT: "right",
  TOP: "top",
  BOTTOM: "bottom",
};

export const POSITION_DIRECTIONS = {
  VERTICAL: "vertical",
  HORIZONTAL: "horizontal",
};

export const TOOLTIP_PADDING = 12;

export const TOOLTIP_ARROW_SIZE = 7;

export const TOOLTIP_TOTAL_PADDING: number = TOOLTIP_PADDING + TOOLTIP_ARROW_SIZE;
