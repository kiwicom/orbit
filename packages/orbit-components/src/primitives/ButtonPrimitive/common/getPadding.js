// @flow
import { rtlSpacing } from "../../../utils/rtl";
import { SIZE_OPTIONS, TOKENS } from "./consts";
import type { GetPadding } from "./getPadding";

const getSpacing: GetPadding = (onlyIcon, iconRight, iconLeft, size, theme) => {
  const wrappedRtl = value => rtlSpacing(value)({ theme });
  if (onlyIcon) return wrappedRtl(theme.orbit.paddingButtonWithoutText);
  const tokens = {
    [TOKENS.paddingButton]: {
      [SIZE_OPTIONS.LARGE]: theme.orbit.paddingButtonLarge,
      [SIZE_OPTIONS.NORMAL]: theme.orbit.paddingButtonNormal,
      [SIZE_OPTIONS.SMALL]: theme.orbit.paddingButtonSmall,
    },
    [TOKENS.paddingButtonWithIcons]: {
      [SIZE_OPTIONS.LARGE]: theme.orbit.paddingButtonLargeWithIcons,
      [SIZE_OPTIONS.NORMAL]: theme.orbit.paddingButtonNormalWithIcons,
      [SIZE_OPTIONS.SMALL]: theme.orbit.paddingButtonSmallWithIcons,
    },
    [TOKENS.paddingButtonWithLeftIcon]: {
      [SIZE_OPTIONS.LARGE]: theme.orbit.paddingButtonLargeWithLeftIcon,
      [SIZE_OPTIONS.NORMAL]: theme.orbit.paddingButtonNormalWithLeftIcon,
      [SIZE_OPTIONS.SMALL]: theme.orbit.paddingButtonSmallWithLeftIcon,
    },
    [TOKENS.paddingButtonWithRightIcon]: {
      [SIZE_OPTIONS.LARGE]: theme.orbit.paddingButtonLargeWithRightIcon,
      [SIZE_OPTIONS.NORMAL]: theme.orbit.paddingButtonNormalWithRightIcon,
      [SIZE_OPTIONS.SMALL]: theme.orbit.paddingButtonSmallWithRightIcon,
    },
  };
  if (iconLeft && iconRight) return wrappedRtl(tokens[TOKENS.paddingButtonWithIcons][size]);
  if (iconLeft && !iconRight) return wrappedRtl(tokens[TOKENS.paddingButtonWithLeftIcon][size]);
  if (!iconLeft && iconRight) return wrappedRtl(tokens[TOKENS.paddingButtonWithRightIcon][size]);
  return wrappedRtl(tokens[TOKENS.paddingButton][size]);
};

export default getSpacing;
