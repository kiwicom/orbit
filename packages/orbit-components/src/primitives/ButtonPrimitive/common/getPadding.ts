import * as React from "react";

import { rtlSpacing } from "../../../utils/rtl";
import { SIZE_OPTIONS, TOKENS } from "./consts";
import { Size } from "../index.d";
import { Theme } from "../../../defaultTheme";

const getSpacing = (
  onlyIcon: boolean,
  iconRight: React.ReactNode,
  iconLeft: React.ReactNode,
  size: Size,
  theme: Theme,
): string => {
  const wrappedRtl = (value: string) => rtlSpacing(value)({ theme });
  if (onlyIcon) return wrappedRtl(theme.orbit.paddingButtonWithoutText);
  const tokens = {
    [TOKENS.paddingButton]: {
      [SIZE_OPTIONS.LARGE]: theme.orbit.paddingButtonLarge,
      [SIZE_OPTIONS.NORMAL]: theme.orbit.paddingButtonNormal,
      [SIZE_OPTIONS.SMALL]: theme.orbit.paddingButtonSmall,
    },
    [TOKENS.paddingButtonWithIcons]: {
      [SIZE_OPTIONS.LARGE]: theme.orbit.paddingButtonLargeWithIcons,
      [SIZE_OPTIONS.NORMAL]: theme.orbit.paddingButtonNormalWithIcons,
      [SIZE_OPTIONS.SMALL]: theme.orbit.paddingButtonSmallWithIcons,
    },
    [TOKENS.paddingButtonWithLeftIcon]: {
      [SIZE_OPTIONS.LARGE]: theme.orbit.paddingButtonLargeWithLeftIcon,
      [SIZE_OPTIONS.NORMAL]: theme.orbit.paddingButtonNormalWithLeftIcon,
      [SIZE_OPTIONS.SMALL]: theme.orbit.paddingButtonSmallWithLeftIcon,
    },
    [TOKENS.paddingButtonWithRightIcon]: {
      [SIZE_OPTIONS.LARGE]: theme.orbit.paddingButtonLargeWithRightIcon,
      [SIZE_OPTIONS.NORMAL]: theme.orbit.paddingButtonNormalWithRightIcon,
      [SIZE_OPTIONS.SMALL]: theme.orbit.paddingButtonSmallWithRightIcon,
    },
  };
  if (iconLeft && iconRight) return wrappedRtl(tokens[TOKENS.paddingButtonWithIcons][size]);
  if (iconLeft && !iconRight) return wrappedRtl(tokens[TOKENS.paddingButtonWithLeftIcon][size]);
  if (!iconLeft && iconRight) return wrappedRtl(tokens[TOKENS.paddingButtonWithRightIcon][size]);

  return wrappedRtl(tokens[TOKENS.paddingButton][size]);
};

export default getSpacing;
