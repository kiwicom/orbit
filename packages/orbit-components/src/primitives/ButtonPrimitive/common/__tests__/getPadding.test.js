// @flow
import getPadding from "../getPadding";
import { SIZE_OPTIONS } from "../consts";
import theme from "../../../../defaultTheme";

describe("ButtonPrimitive: getPadding function", () => {
  it("with both icons, should match snapshot", () => {
    const onlyIcon = false;
    const iconRight = true;
    const iconLeft = true;
    const size = SIZE_OPTIONS.SMALL;
    expect(getPadding(onlyIcon, iconRight, iconLeft, size, theme)).toMatchSnapshot();
  });
  it("with left icon, should match snapshot", () => {
    const onlyIcon = false;
    const iconRight = false;
    const iconLeft = true;
    const size = SIZE_OPTIONS.LARGE;
    expect(getPadding(onlyIcon, iconRight, iconLeft, size, theme)).toMatchSnapshot();
  });
  it("with right icon, should match snapshot", () => {
    const onlyIcon = false;
    const iconRight = true;
    const iconLeft = false;
    const size = SIZE_OPTIONS.NORMAL;
    expect(getPadding(onlyIcon, iconRight, iconLeft, size, theme)).toMatchSnapshot();
  });
  it("with only icon, should match snapshot", () => {
    const onlyIcon = true;
    const iconRight = false;
    const iconLeft = false;
    const size = SIZE_OPTIONS.NORMAL;
    expect(getPadding(onlyIcon, iconRight, iconLeft, size, theme)).toMatchSnapshot();
  });
});
