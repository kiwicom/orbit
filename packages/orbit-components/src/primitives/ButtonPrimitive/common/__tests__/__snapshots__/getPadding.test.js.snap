// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimitive: getPadding function with both icons, should match snapshot 1`] = `"0 8px"`;

exports[`ButtonPrimitive: getPadding function with left icon, should match snapshot 1`] = `"0 28px 0 16px"`;

exports[`ButtonPrimitive: getPadding function with only icon, should match snapshot 1`] = `"0"`;

exports[`ButtonPrimitive: getPadding function with right icon, should match snapshot 1`] = `"0 12px 0 16px"`;
